### PR TITLE
Improve verbosity management

### PR DESCRIPTION
**Pull request by luclaurent**
New mechanism has been added in the whole package to deal with the management of the verbosity. `logging` package has been used to differentiate different kind of messages printed before using `print` method. 

**The new features are**

1. differentiate the type of log (debug, info, warning, error and critical)
2. with the binaries: 
    a. add verbosity options (`-v` for full verbosity and `-s` for silent mode)
    b. since a `.tex`or a `.xml` file is provided an associated log file will be created with suffix `_am2moodle.log` or `_moode2amc.log` (cleared at every run)
    c. the full verbosity is always given in the log file whatever the options are used with the binaries (notice that management of LateXML outputs could be improved due to the fact that LaTeXML uses only stderr)
    
**In the code:**

- many `print` syntaxes have been adapted to the use of `logging`
- a classification of the use of the levels of logging have been considered but can be improved
- one logger is call on every Python file/module but the name of the logger is only visible in debug mode or in log's files
- the current time (more accurate in debug mode), position of the log message in the sources (in debug mode) and level of logging are used on every log lines
- the outputs of LaTeXML is managed using`subprocess.Popen` and two threads (using `from concurrent.futures import ThreadPoolExecutor`) to simultaneously log `stdout` and `stderr` (notice that in my version of LaTeXML (`0.8.5; rev accf4d96`), the software only use `stderr`)
- `-V` argument for binaries has been introduced to show the version number
-  `-v` argument is now used to activate verbosity mode
- tests have been adapted but the loggers are reduced to the minimum (no log file, basic taggings on every lines)

### CODE REVIEW

**Comment by luclaurent** [comment]
Improvements:
* debug in console: keep only level
* debug in file: keep the current
* change warning level for end message to info
* change date-time to only wall-time in seconds

**Comment by luclaurent** [comment]
New features:

- new module for managing verbosity/logging (`customLogging.py`):
- add new verbosity arguments (`-v` or `-vv`)

*New module:*

- possiblity to setup a console logger with a specific verbosity level (basic logging by default) and a silent mode
- possibility to setup a file logger with specific verbosity level (full verbosity by default) and a silent mode
- silent mode overrides all verbosity options
- `-v` option loads verbosity level 1 (0 by default in console) with elapsed time in ms and show debug messages
-  `-vv` option loads verbosity of level 2 (by default in log file) which add full position in the current Python module's tree and add version number of amc2moodle/moodle2amc

File: amc2moodle/amc2moodle/bin/amc2moodle [hunk 1]
```diff
@@ -61,10 +62,16 @@ def run():
                         help='''No use of system temporary directory, use
                                 input directory instead.''',
                         required=False, action="store_true")
-    parser.add_argument("-v", "--version",
+    parser.add_argument("-V", "--version",
                         help='''Show the current version of moodle2amc''',
                         action="version",
                         version="%(prog)s v{version}".format(version=amdlpkg.__version__))
+    parser.add_argument("-v", "--verbose",
+                        help='''Show all log messages in CLI''',
```
Comment (nennigb) [inline amc2moodle/amc2moodle/bin/amc2moodle:70]: add 'use -vv for more verbosity'.

File: amc2moodle/moodle2amc/bin/moodle2amc [hunk 2]
```diff
@@ -51,10 +52,16 @@ def run():
...
                         help='''Check if LaTeX output compile (optional)''',
                         required=False, action="store_true")
-    parser.add_argument("--version",
+    parser.add_argument("-V","--version",
                         help='''Show the current version of moodle2amc''',
                         action="version",
                         version="%(prog)s v{version}".format(version=amdlpkg.__version__))
+    parser.add_argument("-v", "--verbose",
+                        help='''Show all log messages in CLI''',
+                        required=False, action="count")
```
Comment (nennigb) [inline amc2moodle/moodle2amc/bin/moodle2amc:61]: same as before add 'use -vv for more verbosity'

**Review by nennigb — COMMENTED** [review]
Very nice!
It seems that using -vv flag is not present in the command line doc.
If you can edit `amc2moodle` and `moodle2amc` it will be perfect.

File: amc2moodle/amc2moodle/bin/amc2moodle [hunk 3]
```diff
@@ -83,32 +90,67 @@ def run():
     tempDir = not args.notemp
     catname = args.catname
     magic_flag = not(args.magic_flag)
+    silentMode = args.silent
+    verboseMode = args.verbose
+
+    #adapt verbosemode
+    if not verboseMode:
+        verboseMode = 0
```
Comment (nennigb) [inline amc2moodle/amc2moodle/bin/amc2moodle:98]: may be added directly in `parser.add_argument` using ` default=0`

**Comment by luclaurent** [comment]
I improve the code with all your comments/remarks: 

- the use of `-vv` is now proposed in the help
- i added the `--no-log-file` option to deactivate the log file
- i have modified the LaTeX's check process in `moodle2mac`  using `Popen`and a parallel management of stdout and stderr (notice that `pdflatex` only uses `stdout`

**Review by nennigb — APPROVED** [review]
Ok for me!
Let me know if I can merge or if you want to add something else ;-)

**Pull request merged**